internal/querynodev2/delegator/idf_oracle.go

```diff
@@ -106,40 +108,43 @@ type sealedBm25Stats struct {
 	removed   bool
 	segmentID int64
 	ts        time.Time // Time of segemnt register, all segment resgister after target generate will don't remove
-	localPath string
+	localDir  string
+	fieldList []int64 // bm25 field list
 }
 
-func (s *sealedBm25Stats) writeFile(localPath string) (error, bool) {
+func (s *sealedBm25Stats) writeFile(localDir string) (error, bool) {
 	s.RLock()
 
-	if s.removed {
+	if s.removed || !s.inmemory {
 		return nil, true
 	}
 
-	m := make(map[int64][]byte, len(s.bm25Stats))
 	stats := s.bm25Stats
 	s.RUnlock()
 
+	err := os.MkdirAll(localDir, fs.ModePerm)
+	if err != nil {
+		return err, false
+	}
+
 	// RUnlock when stats serialize and write to file
 	// to avoid block remove stats too long when sync distribution
 	for fieldID, stats := range stats {
-		bytes, err := stats.Serialize()
+		file, err := os.Create(path.Join(localDir, fmt.Sprintf("%d.data", fieldID)))
```

Comment:
defer file.Close() to avoid fd leak, and check error

Reply:
> defer file.Close() to avoid fd leak, and check error

Closed, just not use defer, because defer will close after function return. We could be closed earlier here.